addons/mail/static/tests/gif_picker/gif_picker_test.js

```diff
@@ -0,0 +1,125 @@
+    await click("img[data-src='https://media.tenor.com/6uIlQAHIkNoAAAAM/cry.gif']");
+    await click(".o-discuss-gif-picker-popover .fa-arrow-left");
+    assert.containsOnce($, ".o-discuss-gif-categories");
+});
+
+QUnit.test("Make a gif favorite", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    await click("img[data-src='https://media.tenor.com/6uIlQAHIkNoAAAAM/cry.gif']");
+    const firstGif = document.querySelector(".o-discuss-gif .fa-star-o");
+    await click(firstGif);
+    assert.containsOnce($, ".o-discuss-gif .fa-star");
+    await click(".o-discuss-gif-picker-popover .fa-arrow-left");
+    await click(".o-discuss-gif-category:contains(Favorites)");
+    assert.containsOnce($, ".o-discuss-gif");
+});
+
+QUnit.test("Remove a gif favorite", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    await click("img[data-src='https://media.tenor.com/6uIlQAHIkNoAAAAM/cry.gif']");
+    const firstGif = document.querySelector(".o-discuss-gif .fa-star-o");
+    await click(firstGif);
+    await click(".o-discuss-gif-picker-popover .fa-arrow-left");
+    await click(".o-discuss-gif-category:contains(Favorites)");
+    await click(".o-discuss-gif .fa-star");
+    assert.containsNone($, ".o-discuss-gif");
+});
+
+QUnit.test("Chatter should not have the gif button", async (assert) => {
+    const { openFormView, pyEnv } = await start();
+    const partnerId = pyEnv["res.partner"].create({ name: "John Doe" });
+    await openFormView("res.partner", partnerId);
+    await click("button:contains(Log note)");
+    assert.containsNone($, "button[aria-label='Gifs']");
+});
+
+QUnit.test("Composer gif button should open the gif picker", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    assert.containsOnce($, ".o-discuss-gif-picker-popover");
+});
+
+QUnit.test("Mobile: Composer gif button should open the gif picker keyboard", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "channel" });
+    patchUiSize({ size: SIZES.SM });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId, { waitUntilMessagesLoaded: false });
+    await click("span:contains(channel)");
+    await click("button[aria-label='Gifs']");
+    assert.containsNone($, ".o-discuss-gif-picker-popover");
+    assert.containsOnce($, ".o-mail-Composer-gif-picker");
+});
```

Comment:
This test is useless functionally. Assertion on classnames are bad, because you can add classnames anywhere in template and it doesn't mean anything.

I prefer no test than this kind of test, because it's one less (bad) test to maintain.
I don't mind introducing classnames when it's something conceptually important (`o-discuss-GifPicker`. `o-discuss-Gif`).
The popover part would be better as `.popover .o-discuss-GifPicker`.
The case for mobile, you can assert it's not `.popover .o-discuss-GifPicker` but there's a `.o-discuss-Picker`.
Would be fine with `.o-mail-Composer-footer .o-discuss-Picker`, so we know the intent showing in mobile.